### ISSUE TITLE
How to load XR Cardboard on runtime

### ISSUE THREAD

**Issue by ediatreyu** [googlevr/cardboard#53]
Hello before XR management the way to start with "None XR device" an  load "Cardboard" on runtime was like this:

```
 XRSettings.LoadDeviceByName(newDevice);
 yield return null;
 XRSettings.enabled = true;
```

but now I can not find the way to implement this ?,  is it possible with this new Cardboard plugin ?, thanks.

**Comment by CedoSnr** [comment]
Use the following: 

`XRGeneralSettings.Instance.Manager.InitializeLoaderSync();
        XRGeneralSettings.Instance.Manager.StartSubsystems();`

**Comment by ZioCain** [comment]
`XRGeneralSettings.Instance.Manager.InitializeLoaderSync(); XRGeneralSettings.Instance.Manager.StartSubsystems();`
These two don't seem to even exist in the project after including all of the Cardboard plugin, am I doing something wrong?

**Comment by CedoSnr** [comment]
> `XRGeneralSettings.Instance.Manager.InitializeLoaderSync(); XRGeneralSettings.Instance.Manager.StartSubsystems();`
> These two don't seem to even exist in the project after including all of the Cardboard plugin, am I doing something wrong?

These are only needed if you want to initialise Google VR post-launch. You would need to add them to a script somewhere yourself.

**Comment by ZioCain** [comment]
> > `XRGeneralSettings.Instance.Manager.InitializeLoaderSync(); XRGeneralSettings.Instance.Manager.StartSubsystems();`
> > These two don't seem to even exist in the project after including all of the Cardboard plugin, am I doing something wrong?
> 
> These are only needed if you want to initialise Google VR post-launch. You would need to add them to a script somewhere yourself.

That is exactly what I'd like to do, but even tho I installed both Cardboard package and XR Plugin Manager package, I can't make it work, how am I supposed to do that?

**Comment by CedoSnr** [comment]
> That is exactly what I'd like to do, but even tho I installed both Cardboard package and XR Plugin Manager package, I can't make it work, how am I supposed to do that?

Make sure Cardboard XR is enabled under Project Settings -> XR Plug-in Management, but "Initialize XR on Startup is not ticked". Then just put the above code into a script that runs at whatever point you want to initialise XR. 
Bear in mind this won't run in the editor, you'll only see a difference an android device or emulator.

**Comment by ZioCain** [comment]
Ok, I made it!
I now have my function that goes like this:
```
void SetVR(bool active){
    if (active) {
        UnityEngine.XR.Management.XRGeneralSettings.Instance.Manager.InitializeLoaderSync();
        UnityEngine.XR.Management.XRGeneralSettings.Instance.Manager.StartSubsystems();
    }else{
        UnityEngine.XR.Management.XRGeneralSettings.Instance.Manager.StopSubsystems();
    }
}
```
Problem is that now once VR is activated, the app crashes:
```
[XR] [CardboardXrMain]: Display subsystem successfully loaded.
[XR] [CardboardXrMain]: Input subsystem successfully loaded.
[XR] [CardboardXrInputProvider]: Lifecycle initialized
[XR] [CardboardXrInputProvider]: Lifecycle started
[XR] [CardboardXrDisplayProvider]: Initializes Cardboard API.
```
Am I doing something wrong? It crashes right after running that code above...

**Comment by jballoffet** [comment]
As @CedoSnr said, you should use XRGeneralSettings to control the XR subsystems’ lifecycle. For further information, take a look at https://docs.unity3d.com/Packages/com.unity.xr.management@3.2/api/UnityEngine.XR.Management.html.

**Comment by CedoSnr** [comment]
> Am I doing something wrong? It crashes right after running that code above...

Try making a blank project or scene with just the two lines of code mentioned before and have them run in isolation on startup. I can't help you with the rest of your code.

**Comment by JoaoBorks** [comment]
I've submitted the Pull Request googlevr/cardboard-xr-plugin#13 to add this feature to the sample

**Comment by chaosemer** [comment]
This was fixed in v1.4.0.
